### ISSUE TITLE
Build Task : Do not use `Build.BuildNumber` as default. Instead let flutter/user handle it.

### ISSUE THREAD

**Issue by bobwiller** [hey24sheep/azure-flutter-tasks#10]
If using the default build number, Android builds fails with the error:

`buildNumber: 20210920.1 was not a valid integer value.`

For reference, the build command line looks like this:

`/opt/hostedtoolcache/Flutter/2.5.1-stable/linux/flutter/bin/flutter build apk --build-number=20210920.1`

the Task looks like this:

- task: FlutterBuild@0
  inputs:
    target: 'apk'
    projectDirectory: '.'

Is there a different/recommended way to auto-increment Android build numbers?

**Comment by MagnusJohansson** [comment]
Same problem here.
I have not specified the buildNumber parameter, and yet it seems to add it by itself:
My task:
```yaml
  - task: FlutterBuild@0
    inputs:
      target: aab
      projectDirectory: $(projectDirectory)

```

Which executes as:
```
flutter build appbundle --build-number=20210922.2
```

which then generates this error:
```
buildNumber: 20210922.2 was not a valid integer value.
```

How can I make the build task simply take the manifest values from the source?
(As `flutter build appbundle` does by default from the command line)

**Comment by hey24sheep** [comment]
If you read the readme `$(Build.BuildNumber)` is the default value to increment the generated Flutter files (apk/ios) build id.
This is related to the way your build numbers are generated which is a decimal value instead of an integer.

Here is my task :
```yaml
steps:
- task: Hey24sheep.flutter.flutter-build.FlutterBuild@0
  displayName: 'Flutter Build apk'
  inputs:
    projectDirectory: $(projectDirectory)
    debugMode: true
    buildFlavour: development
    entryPoint: 'lib/main_dev.dart'
    apkTargetPlatform: 'android-arm64'
```

Here is my generated command from the above task - `build apk "--target-platform=android-arm64" "--build-number=15530" "--flavor=development" "--target=lib/main_dev.dart"` as you can see my build number is an 'Int'. 

As to the latest Flutter update regarding [Flutter versioning](https://medium.com/karlmax-berlin/versioning-with-flutter-299869e68af4) where you can use `pubspec.yaml` to auto increment versions. I will make it default to none.

**Comment by hey24sheep** [comment]
Fixed in fe53bd0. Now `Build task` is truely optional so it can be null and also split decimal value.
